torch/fx/experimental/proxy_tensor.py

```diff
@@ -60,13 +60,28 @@ def wrap_with_proxy(e, proxy):
         return inner_res
 
 
+def maybe_disable_fake_tensor_mode():
+    # TODO: figure out if this API generally makes sense and bake it into the
+    # library
+    mb_fake_mode = torch._C._get_torch_dispatch_mode()
+    if isinstance(mb_fake_mode, FakeTensorMode):
```

Comment:
What's the problem with FakeTensor Mode here and why don't we see this come up with other modes? In other words, why do we need to do this over `no_dispatch`?

Reply:
A few things. First, `no_dispatch` doesn't work, as it only disables subclass dispatch and not mode dispatch. OK, what about just disabling all mode dispatch? This is less compositional; suppose there is some debug mode (like a logging mode) that is running under proxy tensor; we'd disable everything even if we only wanted to disable one operation.

Now, why do we need to disable fake tensor mode at all? The problem is that I want to do constant prop, and to do constant prop, I need to do actual compute, but if a fake tensor mode is ambiently on then I cannot because anything I do to try to compute the tensor will turn into fake tensors.